### PR TITLE
Impove handling of cli files

### PR DESCRIPTION
**Pull request by ZanderBrown**
Further work for #409 

Whilst this probably does resolve that issue it does resolve an issue that issues caused so-to-speak

### CODE REVIEW

File: mu/app.py [hunk 1]
```diff
@@ -104,9 +104,18 @@ def run():
...
-    # capture the filename passed by the os, if there was one
-    passed_filename = sys.argv[1] if len(sys.argv) > 1 else None
-    editor.restore_session(os.path.abspath(passed_filename))
+    # Restore the previous session
+    editor.restore_session(without_empty=len(sys.argv[1:]) > 0)
+    # capture the filename passed by the os, if there were any
+    for passed_filename in sys.argv[1:]:
+        try:
+            # abspath will fail for non-paths
+            editor.direct_load(os.path.abspath(passed_filename))
```
Comment (ntoll) [inline mu/app.py]: hmmm.... I'm not sure about this approach. The code in `app.py` is essentially there to instantiate things and connect them together. This feels like logic that shouldn't be here (i.e. it should be within the editor, and all the editor needs are a bunch of paths for the passed in files). Does this make sense?

Comment (ZanderBrown) [reply mu/app.py]: Yep, I'll have a look at rearranging that

**Comment by ZanderBrown** [comment]
Sorry that took a while

File: mu/app.py [hunk 2]
```diff
@@ -104,9 +104,10 @@ def run():
     # Setup the window.
     editor_window.closeEvent = editor.quit
     editor_window.setup(editor.debug_toggle_breakpoint, editor.theme)
-    # capture the filename passed by the os, if there was one
-    passed_filename = sys.argv[1] if len(sys.argv) > 1 else None
-    editor.restore_session(os.path.abspath(passed_filename))
+    # Restore the previous session
+    editor.restore_session(without_empty=len(sys.argv[1:]) > 0)
```
Comment (ntoll) [inline mu/app.py]: Hmmm... so why pass this flag at all. Why not pass in `sys.argv[1:]` and let `restore_session` handle it..? Happy to be talked round, but it feels like there should be one call to `restore_session` with the passed in files and then it'll work out what to do with it given the previous state of the session and the new files. Does that make sense?

**Comment by ntoll** [comment]
No need to apologies for anything @ZanderBrown... I'm just incredibly grateful that people are taking the time to improve Mu. :-)

File: mu/app.py (hunk 2 above)
Comment (ZanderBrown) [reply mu/app.py]: I had them separate in case we wanted to do single instance things but I'll move the call inside `restore_session`

**Comment by ntoll** [comment]
Thanks! :-)

**Pull request merged**